pkg/features/pilot/pilot.go

```diff
@@ -114,16 +114,19 @@ var (
 		return time.Second * time.Duration(duration)
 	}
 
+	// EnableCDSPrecomputation provides an option to enable precomputation
+	// of CDS output for all namespaces at the start of a push cycle.
+	// While it reduces CPU, it comes at the cost of increased memory usage
```

Comment:
That's the part that worries me. "reduced CPU at the cost of memory usage" means we generate more GC, and there is at least some 'hit' on the cache. 

The behavior under pressure (where CPU can't cope) is really bad - if it's a 'steady state' increase in memory, because we cache more, it is still better than allocating the same memory on every request. 

This is the kind of change that requires a lot of testing under heavy load - if Mandar can confirm that under stress ( i.e. when GC starts to play the dominant role ) we are ok, and that the change doesn't affect the scalability limit - I'm good. 

But all scale and perf testing so far has been done with the cache enabled, and we are extremely late in the release.

Reply:
Look at mandar's comments in this Pr. He tested on large clusters, ran it for a while.. memory usage is low overall.